src/implementation.cpp

```diff
@@ -118,7 +118,7 @@ const implementation *available_implementation_list::detect_best_supported() con
     uint32_t required_instruction_sets = impl->required_instruction_sets();
     if ((supported_instruction_sets & required_instruction_sets) == required_instruction_sets) { return impl; }
   }
-  return &unsupported_singleton;
+  return &unsupported_singleton; // this should never happen?
```

Comment:
So, I've actually had in the back of my mind that this is a really appropriate place for an abort(), and was planning to come add it in and remove unsupported_singleton. It could happen if someone turned SIMDJSON_IMPLEMENTATION_FALLBACK=OFF, but I think we could say "whatever" to that.